src/internal_rep.rs

```diff
@@ -1539,6 +1552,316 @@ fn call_to_fc_rules<'a>(
+    pub file_type: Option<FileType>,
+    pub context: Context<'a>,
+    #[derivative(PartialEq = "ignore")]
+    #[derivative(PartialOrd = "ignore")]
+    #[derivative(Ord = "ignore")]
+    pub file: SimpleFile<String, String>,
+    #[derivative(PartialEq = "ignore")]
+    #[derivative(PartialOrd = "ignore")]
+    #[derivative(Ord = "ignore")]
+    pub func_call: FuncCall,
+}
+impl Eq for FileSystemContextRule<'_> {}
+
+impl FileSystemContextRule<'_> {
+    fn get_renamed_statement(&self, renames: &BTreeMap<String, String>) -> Self {
+        FileSystemContextRule {
+            fscontext_type: self.fscontext_type.clone(),
+            fs_name: self.fs_name.clone(),
+            path: self.path.clone(),
+            file_type: self.file_type,
+            context: self.context.get_renamed_context(renames),
+            file: self.file.clone(),
+            func_call: self.func_call.clone(),
+        }
+    }
+}
+
+impl TryFrom<&FileSystemContextRule<'_>> for sexp::Sexp {
+    type Error = ErrorItem;
+
+    fn try_from(f: &FileSystemContextRule) -> Result<sexp::Sexp, ErrorItem> {
+        match f.fscontext_type {
+            FSContextType::XAttr | FSContextType::Task | FSContextType::Trans => Ok(list(&[
+                atom_s("fsuse"),
+                Sexp::Atom(Atom::S(f.fscontext_type.to_string())),
+                atom_s(f.fs_name.trim_matches('"')),
+                Sexp::from(&f.context),
+            ])),
+            FSContextType::GenFSCon => {
+                if let Some(p) = &f.path {
+                    if let Some(file_type) = &f.file_type {
+                        // TODO add secilc check here. Right now our github pipeline
+                        // supports an older version of secilc.  So to get things moving forward
+                        // we are forcing the old behavior.  The new behavior has been tested locally.
+                        // REMEMBER TO UPDATE THE TESTS
+                        // if secilc/libsepol version is new enough {
+                        if false {
+                            return Ok(list(&[
+                                atom_s("genfscon"),
+                                atom_s(f.fs_name.trim_matches('"')),
+                                atom_s(p.as_ref()),
+                                Sexp::Atom(Atom::S(file_type.to_string())),
+                                Sexp::from(&f.context),
+                            ]));
+                        }
+                    }
+                    // We are purposefully falling through without an else to
+                    // reduce redundant lines of code
+                    Ok(list(&[
+                        atom_s("genfscon"),
+                        atom_s(f.fs_name.trim_matches('"')),
+                        atom_s(p.as_ref()),
+                        Sexp::from(&f.context),
+                    ]))
+                } else {
+                    // We should never get here since we are defaulting to "/"
```

Comment:
If we should never hit this in normal operation, should it be an InternalError instead?